libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -681,143 +752,143 @@ bool AC_WPNav::set_spline_destination_loc(const Location& destination, const Loc
+/// set_spline_destination_NEU_m waypoint using position vector (distance from ekf origin in m)
+///     is_terrain_alt should be true if destination_neu_m.z is a desired altitude above terrain (false if its desired altitudes above ekf origin)
+///     next_destination_neu_m should be set to the next segment's destination
+///     next_is_terrain_alt should be true if next_destination_neu_m.z is a desired altitude above terrain (false if its desired altitudes above ekf origin)
+///     next_destination_neu_m.z  must be in the same "frame" as destination_neu_m.z (i.e. if destination is a alt-above-terrain, next_destination should be too)
+bool AC_WPNav::set_spline_destination_NEU_m(const Vector3f& destination_neu_m, bool is_terrain_alt, const Vector3f& next_destination_neu_m, bool next_is_terrain_alt, bool next_is_spline)
 {
     // re-initialise if previous destination has been interrupted
     if (!is_active() || !_flags.reached_destination) {
-        wp_and_spline_init_cm(_wp_desired_speed_ne_cms);
+        wp_and_spline_init_m(_wp_desired_speed_ne_ms);
     }
 
     // update spline calculators speeds and accelerations
-    _spline_this_leg.set_speed_accel(_pos_control.get_max_speed_NE_cms(), _pos_control.get_max_speed_up_cms(), _pos_control.get_max_speed_down_cms(),
-                                     _pos_control.get_max_accel_NE_cmss(), _pos_control.get_max_accel_U_cmss());
+    _spline_this_leg.set_speed_accel(_pos_control.get_max_speed_NE_ms(), _pos_control.get_max_speed_up_ms(), _pos_control.get_max_speed_down_ms(),
+                                     _pos_control.get_max_accel_NE_mss(), _pos_control.get_max_accel_U_mss());
 
     // calculate origin and origin velocity vector
-    Vector3f origin_vector_neu_cm;
+    Vector3f origin_vector_neu_m;
     if (is_terrain_alt == _is_terrain_alt) {
         if (_flags.fast_waypoint) {
             // calculate origin vector
             if (_this_leg_is_spline) {
                 // if previous leg was a spline we can use destination velocity vector for origin velocity vector
-                origin_vector_neu_cm = _spline_this_leg.get_destination_vel();
+                origin_vector_neu_m = _spline_this_leg.get_destination_vel();
             } else {
                 // use direction of the previous straight line segment
-                origin_vector_neu_cm = _destination_neu_cm - _origin_neu_cm;
+                origin_vector_neu_m = _destination_neu_m - _origin_neu_m;
             }
         }
 
         // use previous destination as origin
-        _origin_neu_cm = _destination_neu_cm;
+        _origin_neu_m = _destination_neu_m;
     } else {
 
         // use previous destination as origin
-        _origin_neu_cm = _destination_neu_cm;
+        _origin_neu_m = _destination_neu_m;
 
         // get current alt above terrain
-        float origin_terr_offset_u_cm;
-        if (!get_terrain_offset_cm(origin_terr_offset_u_cm)) {
+        float origin_terr_offset;
```

Comment:
this name change is intentional?  .. or maybe it should be "float origin_terr_offset_u_m"?